crypto/stark/examples/pedersen_merkle/inputs.rs

```diff
@@ -13,39 +13,43 @@ pub struct Claim {
 #[derive(PartialEq, Clone)]
 #[cfg_attr(feature = "std", derive(Debug))]
 pub struct Witness {
-    pub directions: Vec<bool>,
-    pub path:       Vec<FieldElement>,
+    pub path: Vec<(bool, FieldElement)>,
+    pub leaf: FieldElement,
+    pub root: FieldElement,
 }
 
-impl From<&Claim> for Vec<u8> {
-    fn from(claim: &Claim) -> Self {
-        let mut bytes: Self = vec![];
-        bytes.extend_from_slice(&claim.path_length.to_be_bytes());
-        bytes.extend_from_slice(&claim.root.as_montgomery().to_bytes_be());
-        bytes.extend_from_slice(&claim.leaf.as_montgomery().to_bytes_be());
-        bytes
+impl Witness {
+    pub fn new(leaf: FieldElement, path: Vec<(bool, FieldElement)>) -> Witness {
+        let root = path
+            .iter()
+            .fold(leaf.clone(), |leaf, (direction, sibling)| {
+                if *direction {
+                    merkle_hash(sibling, &leaf)
+                } else {
+                    merkle_hash(&leaf, sibling)
+                }
+            });
+        Witness { leaf, path, root }
     }
 }
 
-impl Claim {
-    pub fn from_leaf_witness(leaf: FieldElement, witness: &Witness) -> Self {
-        let mut root = leaf.clone();
-        for (direction, sibling) in witness.directions.iter().zip(witness.path.iter()) {
-            root = if *direction {
-                merkle_hash(sibling, &root)
-            } else {
-                merkle_hash(&root, sibling)
-            }
-        }
+impl From<&Witness> for Claim {
+    fn from(witness: &Witness) -> Self {
         Claim {
             path_length: witness.path.len(),
-            leaf,
-            root,
+            leaf:        witness.leaf.clone(),
+            root:        witness.root.clone(),
         }
     }
+}
 
-    pub fn verify(&self, witness: &Witness) {
-        assert_eq!(self, &Claim::from_leaf_witness(self.leaf.clone(), witness));
+impl From<&Claim> for Vec<u8> {
+    fn from(claim: &Claim) -> Self {
+        let mut bytes: Self = vec![];
+        bytes.extend_from_slice(&claim.path_length.to_be_bytes());
+        bytes.extend_from_slice(&claim.root.as_montgomery().to_bytes_be());
+        bytes.extend_from_slice(&claim.leaf.as_montgomery().to_bytes_be());
```

Comment:
The path: .as_montgomery().to_bytes_be() is used quite a bit throughout our code, and I think it would read better as a path like .as_montgomery_bytes(). That change is out of scope for this pr though.